server/store/datastore/redirection_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastore
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/woodpecker-ci/woodpecker/server/model"
+)
+
+func TestGetRedirection(t *testing.T) {
+	store, closer := newTestStore(t, new(model.Redirection))
+	defer closer()
+
+	redirection := &model.Redirection{
+		RepoID:   1,
+		FullName: "foo/bar",
+	}
+	if err := store.CreateRedirection(redirection); err != nil {
+		t.Errorf("Unexpected error: insert redirection: %s", err)
+		return
+	}
+	redirectionFromStore, err := store.GetRedirection("foo/bar")
+	assert.NoError(t, err)
+	assert.NotNil(t, redirectionFromStore)
+	assert.Equal(t, redirection.RepoID, redirectionFromStore.RepoID)
+	_, err = store.GetRedirection("foo/baz")
+	assert.ErrorIs(t, err, RecordNotExist)
+}
+
+func TestCreateRedirection(t *testing.T) {
+	store, closer := newTestStore(t, new(model.Redirection))
+	defer closer()
+
+	redirection := &model.Redirection{
+		RepoID:   1,
+		FullName: "foo/bar",
+	}
+	assert.NoError(t, store.CreateRedirection(redirection))
+}
+
+func TestHasRedirectionForRepo(t *testing.T) {
+	store, closer := newTestStore(t, new(model.Redirection))
+	defer closer()
+
+	redirection := &model.Redirection{
+		RepoID:   1,
+		FullName: "foo/bar",
+	}
+	if err := store.CreateRedirection(redirection); err != nil {
```

Comment:
try to create a redirect that should not work and check if it errors too

good tests not only test against success but also against expected errors